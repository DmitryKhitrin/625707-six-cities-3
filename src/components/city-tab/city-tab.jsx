import React, {useCallback} from "react";
import {PropTypes} from "prop-types";

export const CityTab = ({cityName, setCity, getOffers, isActive}) => {
  const onClick = useCallback(() => {
    setCity(cityName);
    getOffers(cityName);
  }, [cityName, setCity, getOffers]);
  return (
    <li className="locations__item" onClick={onClick}>
      <a
        className={`locations__item-link tabs__item ${
          isActive ? `tabs__item--active` : ``
        }`}
        href="#"
      >
        <span>{cityName}</span>
      </a>
    </li>
  );
};

CityTab.propTypes = {
  cityName: PropTypes.string.isRequired,
  setCity: PropTypes.func.isRequired,
  getOffers: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired
};
