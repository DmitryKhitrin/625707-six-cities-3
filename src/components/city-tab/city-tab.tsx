import React, {FC, useCallback, memo} from "react";

type Props = {
  cityName: string;
  setCity: (cityName: string) => void;
  isActive: boolean;
}

const CityTab: FC<Props> = ({cityName, setCity, isActive}) => {
  const onClick = useCallback(() => {
    setCity(cityName);
  }, [cityName, setCity]);
  return (
    <li className="locations__item" onClick={onClick}>
      <a
        className={`locations__item-link tabs__item ${
          isActive ? `tabs__item--active` : ``
        }`}
        href="#"
      >
        <span>{cityName}</span>
      </a>
    </li>
  );
};

const MemoizedCityTab = memo(CityTab);
export {MemoizedCityTab as CityTab};
