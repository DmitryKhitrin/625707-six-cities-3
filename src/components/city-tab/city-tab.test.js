import React from "react";
import renderer from "react-test-renderer";
import {CityTab} from "./city-tab.jsx";
it(`<CityTab /> should render CityTab.`, () => {

  const tree = renderer
    .create(
        <CityTab
          cityName={`Moscow`}
          setCity={jest.fn()}
          isActive={false}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
