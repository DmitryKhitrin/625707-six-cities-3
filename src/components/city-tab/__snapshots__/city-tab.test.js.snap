// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityTab /> should render CityTab. 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item "
    href="#"
  >
    <span>
      Moscow
    </span>
  </a>
</li>
`;
